src/home/NotificationCenter.tsx

```diff
@@ -0,0 +1,118 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { LayoutChangeEvent, StyleSheet, Text, View } from 'react-native'
+import Animated, { useAnimatedScrollHandler, useSharedValue } from 'react-native-reanimated'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import type { Notification } from 'src/home/NotificationBox'
+import { useNotifications } from 'src/home/NotificationBox'
+import ThumbsUpIllustration from 'src/icons/ThumbsUpIllustration'
+import { Screens } from 'src/navigator/Screens'
+import useScrollAwareHeader from 'src/navigator/ScrollAwareHeader'
+import { StackParamList } from 'src/navigator/types'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+type NotificationsProps = NativeStackScreenProps<StackParamList, Screens.NotificationCenter>
+
+const renderItem = ({ item }: { item: Notification }) => {
+  return (
+    <View testID={`NotificationView/${item.id}`} key={item.id} style={styles.listItem}>
+      {item.element}
+    </View>
+  )
+}
+
+const keyExtractor = (item: Notification) => item.id
+
+const renderItemSeparator = () => <View style={styles.itemSeparator} />
```

Comment:
i usually see functions like these inside the component - i don't have a problem with them out here, but could you remind me of the difference / benefits? it's nice that these functions don't get recreated every re-render of the Notifications component, and they don't need anything specifically inside the Notifications component so there's no _need_ for them to live inside Notifications. are there other considerations?

Reply:
no, i don't know any new aspects besides that the functions don't get recreated every re-render.
i've noticed that having at least FlatList's `keyExtractor` outside the component is quite frequent among the codebase, so I considered it a common practice 😅 , for instance:
[CeloNewsFeed.tsx](https://github.com/valora-inc/wallet/blob/0d4dc3c04834ef5893b15342f186cf3a1f08bb0b/src/exchange/CeloNewsFeed.tsx#L29-L39)
[SelectLocalCurrency.tsx](https://github.com/valora-inc/wallet/blob/0d4dc3c04834ef5893b15342f186cf3a1f08bb0b/src/localCurrency/SelectLocalCurrency.tsx#L17-L19)
[SelectCountry.tsx](https://github.com/valora-inc/wallet/blob/0d4dc3c04834ef5893b15342f186cf3a1f08bb0b/src/onboarding/registration/SelectCountry.tsx#L17)
[Language.tsx](https://github.com/valora-inc/wallet/blob/0d4dc3c04834ef5893b15342f186cf3a1f08bb0b/src/language/Language.tsx#L25-L27)

i don't believe that there is a substantial performance win, so if placing these functions within the component will contribute to code readability and portability, i have no objections. what do you think?

Reply:
I think, for readability it's worth not to tear them apart. 
grouped all inside the component 📦